eng/versioning/external_dependencies.txt

```diff
@@ -81,7 +81,7 @@ org.apache.avro:avro-maven-plugin;1.11.0
 org.apache.commons:commons-compress;1.22
 org.apache.commons:commons-lang3;3.12.0
 org.apache.httpcomponents:httpclient;4.5.14
-org.apache.kafka:kafka-clients;3.3.2
+org.apache.kafka:kafka-clients;3.4.0
```

Comment:
Please check the spring boot's bom, to see which version it includes in the latest version of 3.0.x and 3.1.x. IIRC, spring bom will only upgrade the patch version of a 3rd party library in a Spring Boot minor version.

Reply:
Spring Boot 3.1.0 upgrade kafka-clients to 3.4.0, shall we waiting this in `feature/spring-boot-3` branch?